src/vcpkg/vcpkgpaths.cpp

```diff
@@ -672,9 +679,22 @@ namespace vcpkg
                                                        m_pimpl->m_config_dir,
                                                        *this);
 
+            translateRelativePaths();
```

Comment:
```suggestion            
            auto make_relative_to_manifest = [&](std::string& overlay_path) {
                auto p = Path(overlay_path);
                if (p.is_relative())
                {
                    if (auto manifest_directory = get_manifest_directory().get())
                    {
                        auto res = *manifest_directory / p;
                        overlay_path = res.generic_u8string();
                    }
                }
            };

            auto& config_ports = get_configuration().config.overlay_ports;
            auto& config_triplets = get_configuration().config.overlay_triplets;
            Util::transform(config_ports, make_relative_to_manifest);
            Util::transform(config_triplets, make_relative_to_manifest);
```

maybe something like this is cleaner?